src/transformers/processing_utils.py

```diff
@@ -648,21 +649,23 @@ def from_args_and_dict(cls, args, processor_dict: Dict[str, Any], **kwargs):
         processor_dict = processor_dict.copy()
         return_unused_kwargs = kwargs.pop("return_unused_kwargs", False)
 
-        # Unlike image processors or feature extractors whose `__init__` accept `kwargs`, processor don't have `kwargs`.
-        # We have to pop up some unused (but specific) arguments to make it work.
+        # We have to pop up some unused (but specific) kwargs and then validate that it doesn't contain unused kwargs
+        # If we don't pop, some specific kwargs will raise a warning
         if "processor_class" in processor_dict:
             del processor_dict["processor_class"]
 
         if "auto_map" in processor_dict:
             del processor_dict["auto_map"]
 
+        unused_kwargs = cls.validate_init_kwargs(processor_config=processor_dict, valid_kwargs=cls.valid_kwargs)
         processor = cls(*args, **processor_dict)
 
         # Update processor with kwargs if needed
         for key in set(kwargs.keys()):
             if hasattr(processor, key):
                 setattr(processor, key, kwargs.pop(key))
 
+        unused_kwargs.update(kwargs)
```

Comment:
Shouldn't this be kwargs.update(unused_kwargs) given we return kwargs on L671?

Reply:
yeah, right!